### PR TITLE
fix(slider): value-changed event is not fired

### PR DESCRIPTION
**Pull request by Jidapa-Pai**
## Description
Fixes # (issue)
Step to reproduce:

1. Programmatically set `value` by Javascript. e.g. set `value` from 0 to 10.
2. Drag the slider back to 0.
Result: The `value-changed` event is not fired. Try to reproduce here https://codepen.io/jidapa-pai/pen/XWeQdeO

The problem is when user programmatically set `value` the `valuePrevious` which is internal value didn't update so when drag the slider back to previous value, it not detect value change and not fires the `value-changed` event.
So, the solution is to update `valuePrevious` when the `value` was programmatically set.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: packages/elements/src/slider/__test__/slider.test.js [hunk 1]
```diff
@@ -1358,6 +1358,119 @@ describe('slider/Slider', () => {
...
+        // Darg end
+        setTimeout(() => window.dispatchEvent(new MouseEvent('mouseup', { clientX: dragValuePositionStart, clientY: 0 })));
+        await oneEvent(window, 'mouseup');
+        expect(el.dragging).to.be.false;
+        expect(el.value).to.equal(calculateValue(el, dragValuePositionStart).toString());
+
+        // Check call fire event
+        expect(callCountValue).to.equal(1);
+      });
+      it('Event from-changed should fires when from property was set via api and drag the slider back to previous value', async () => {
```
Comment (Sakchai-Refinitiv) [inline packages/elements/src/slider/__test__/slider.test.js:1397]: Making the line space between the test cases improves code readability.

File: packages/elements/src/slider/__test__/slider.test.js [hunk 2]
```diff
@@ -1358,6 +1358,119 @@ describe('slider/Slider', () => {
         expect(toEventFiredCount, 'Event "to-changed" should not fired').to.equal(0);
         expect(slider.to).to.equal('90');
       });
+
+      // jira ELF-1729
```
[the remark this answers is not in the document]
Comment (Theeraphat-Sorasetsakul) [reply packages/elements/src/slider/__test__/slider.test.js:1362]: I think no. I will remove it.

File: packages/elements/src/slider/__test__/slider.test.js [hunk 3]
```diff
@@ -1358,6 +1358,119 @@ describe('slider/Slider', () => {
...
+        // Darg end
+        setTimeout(() => window.dispatchEvent(new MouseEvent('mouseup', { clientX: dragPositionStart, clientY: 0 })));
+        await oneEvent(window, 'mouseup');
+        expect(el.dragging).to.be.false;
+        expect(el.from).to.equal(calculateValue(el, dragPositionStart).toString());
+
+        // Check call fire event
+        expect(callCountValue).to.equal(1);
+      });
+      it('Event to-changed should fires when to property was set via api and drag the slider back to previous value', async () => {
```
Comment (Sakchai-Refinitiv) [inline packages/elements/src/slider/__test__/slider.test.js:1435]: Please add line space

File: packages/elements/src/slider/index.ts [hunk 4]
```diff
@@ -917,6 +917,11 @@ export class Slider extends ControlElement {
         }
       }
     }
+
+    if (!this.dragging) {
+      // Update internal `valuePrevious` when `value` was programatically set by user.
+      this.valuePrevious = this.value;
+    }
```
Comment (Sakchai-Refinitiv) [inline packages/elements/src/slider/index.ts:924]: Why does this change add to the end of the function instead of the beginning?

**Review by Sakchai-Refinitiv — APPROVED** [review]

**Comment by Sakchai-Refinitiv** [comment]
@TremayneChrist @goremikins Do you want to skim this PR before merging?

**Comment by goremikins** [comment]
I do not understand what is going on there and why as the code looks very messy :) but my guess is that the `value-changed` event should be fired not on `blur`, but when the drag is released.

**Comment by Sakchai-Refinitiv** [comment]
> I do not understand what is going on there and why as the code looks very messy :) but my guess is that the `value-changed` event should be fired not on `blur`, but when the drag is released.

If you think the code should be refactored, you can create a new story we can do this in next sprint.

**Comment by Theeraphat-Sorasetsakul** [comment]
> I do not understand what is going on there and why as the code looks very messy :) but my guess is that the `value-changed` event should be fired not on `blur`, but when the drag is released.

I agreed the code like messy to read.
`value-changed` event is fired when drag is released. `onBlur` is for number-field.

**Pull request merged**